src/game/Laptop/IMP_Confirm.cc

```diff
@@ -220,12 +222,45 @@ static void BtnIMPConfirmYes(GUI_BUTTON *btn, INT32 reason)
 		}
 
 		// line moved by CJC Nov 28 2002 to AFTER the check for money
-		LaptopSaveInfo.fIMPCompletedFlag = TRUE;
+		LaptopSaveInfo.fIMPCompletedFlag = AddCharacterToPlayersTeam();
+		if (!LaptopSaveInfo.fIMPCompletedFlag) return; // only if merc hiring failed: no charge, give it another go
+
+		SOLDIERTYPE* const pSoldier = FindSoldierByProfileID(PLAYER_GENERATED_CHARACTER_ID + LaptopSaveInfo.iVoiceId);
+		if (!pSoldier) return;
+
+		if (fLoadingCharacterForPreviousImpProfile && gamepolicy(imp_load_keep_inventory))
+		{
+			IMPSavedProfileLoadInventory(gMercProfiles[PLAYER_GENERATED_CHARACTER_ID + LaptopSaveInfo.iVoiceId].zNickname.c_str(), pSoldier);
+		}
+
+		if (pSoldier->ubID == 0 && FindObj(pSoldier, LETTER) == NO_SLOT)
```

Comment:
Or just drop it directly.

Reply:
The ideal would be to use JSON to export/import to abstract from the compiler-struct memory format. This is not difficult. A text file would make it easier to edit for creating mod scenarios such as the intended "All you need is kill".

The data was placed in an unrelated field because it was an easy hack.

Reply:
The letter placement is not ideal and does not account for the possible case where all slots are filled. If you know how to drop the letter on the map then that would be a possible fix. Can items on the ground be destroyed?

Reply:
I checked it and it's too early to fall back to dropping it to the ground. Haven't had much time to look at all this again yet — will integrate your fixes then, thanks.